raft/fsm.cc

```diff
@@ -487,7 +487,7 @@ void fsm::maybe_commit() {
         } else {
             auto lp = leader_state().tracker.find(_my_id);
             if (lp == nullptr || !lp->can_vote) {
-                logger.trace("maybe_commit[{}]: stepping down as leader", _my_id);
+                logger.info("maybe_commit[{}]: stepping down as leader", _my_id);
```

Comment:
Both of those logs are ugly as they print an internal information not so interested for the regular consumer (but useful to whoever ties to debug a raft issue itself). Raft is a library and as such should be as silent as possible. 

We already can get the information when a node becomes a leader from the topology coordinator output.

Reply:
Let's discuss this on #14524 where I proposed the change.

Reply:
I agree that the logs are ugly and contain internal information.

I think we should keep these as `trace`, instead having separate higher-level logs that don't mention internal function names but are easier for the eyes of a database admin.

BTW. @patjed41 I don't think the logs you marked as INFO cover all cases of losing leadership, they only cover the case of stepping down, not the case where a server notices that there is already a higher-termed leader and becomes a follower, or when it triggers being a candidate due to not getting responses for too long.